### PR TITLE
[RW-6035][risk=no] Move Concept Search to separate route

### PR DESCRIPTION
**Pull request by dolbeew**
- Move concept search page to a separate route from concept homepage
- Cleanup unused code from old concept search ui

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] I have run and tested this change locally
- [x] I have run the E2E tests on ths change against my local UI + API server with `yarn test-local`

### CODE REVIEW

File: ui/src/app/pages/data/criteria-search.tsx [hunk 1]
```diff
@@ -299,20 +298,20 @@ export const CriteriaSearch = fp.flow(withUrlParams(), withCurrentWorkspace())(c
   }
 
   render() {
-    const {backFn, cohortContext, conceptSearchTerms, selectedSurvey, source} = this.props;
+    const {backFn, cohortContext, conceptSearchTerms} = this.props;
```
Comment (NehaBroad) [inline ui/src/app/pages/data/criteria-search.tsx:301]: nit: can we have cohortContext:{source, selectedSurvey} for readability?

**Comment by NehaBroad** [comment]
Since slack is not working i'll share the video here. There is an issue to edit/add selections to existing cohorts (just cohort not concept set)

https://user-images.githubusercontent.com/34481816/103559231-cbb12a80-4e83-11eb-9e56-9fc13c70370d.mov

File: ui/src/app/pages/data/criteria-search.tsx (hunk 1 above)
Comment (dolbeew) [reply ui/src/app/pages/data/criteria-search.tsx]: Done

**Review by dolbeew — COMMENTED** [review]
Nice catch on editing cohorts. I had changed the 'criteria' source prop value to 'cohort' since it made more sense and just missed a few instances. Seems to be working now.

**Review by NehaBroad — APPROVED** [review]

**Pull request merged**